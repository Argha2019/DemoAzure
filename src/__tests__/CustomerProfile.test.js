import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CustomerProfile from '../components/pages/CustomerProfile.jsx';
import {dateValidator} from '../components/pages/CustomerProfile.jsx';
import Cookies from 'js-cookie';

describe('Customer Profile', () => {
    it('render component', () => {
        const wrapper = shallow(<CustomerProfile />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('on change of firstName field', () => {
        const event = {
            preventDefault: () => { },
            target: {
                name: "firstName",
                value: "Yashika",
                readOnly: false,
            }
        };
        const wrapper = shallow(<CustomerProfile />);
        wrapper.setState({
            edit: true,
            readOnly: false,

        });
        const input = wrapper.find('.firstName');

        input.simulate('change', event);
        expect(input.length).toBe(1);
        expect(wrapper.state('firstName')).toBe("Yashika");
    });



    it('on change of lastName field', () => {
        const event = {
            preventDefault: () => { },
            target: {
                name: "lastName",
                value: "Gupta",
            }
        };
        const wrapper = shallow(<CustomerProfile />);
        wrapper.setState({
            edit: true,
            readOnly: false,

        });
        const input = wrapper.find('.lastName');

        input.simulate('change', event);
        expect(input.length).toBe(1);
        expect(wrapper.state('lastName')).toBe("Gupta");
    });

    it('on change of phone field', () => {
        const event = {
            preventDefault: () => { },
            target: {
                name: "phone",
                value: "0000000000",
            }
        };
        const wrapper = shallow(<CustomerProfile />);
        wrapper.setState({
            edit: true,
            readOnly: false,

        });
        const input = wrapper.find('.phone');

        input.simulate('change', event);
        expect(input.length).toBe(1);
        expect(wrapper.state('phone')).toBe("0000000000");
    });

    it('on change of mail field', () => {
        const event = {
            preventDefault: () => { },
            target: {
                name: "mail",
                value: "yashi@123",
            }
        };
        const wrapper = shallow(<CustomerProfile />);
        wrapper.setState({
            edit: true,
            readOnly: false,

        });
        const input = wrapper.find('.mail');

        input.simulate('change', event);
        expect(input.length).toBe(1);
        expect(wrapper.state('mail')).toBe("yashi@123");
    });

    it('on change of addressLine1 field', () => {
        const event = {
            preventDefault: () => { },
            target: {
                name: "addressLine1",
                value: "abcd",
            }
        };
        const wrapper = shallow(<CustomerProfile />);
        wrapper.setState({
            edit: true,
            readOnly: false,

        });
        const input = wrapper.find('.addressLine1');

        input.simulate('change', event);
        expect(input.length).toBe(1);
        expect(wrapper.state('addressLine1')).toBe("abcd");
    });

    it('on change of addressLine2 field', () => {
        const event = {
            preventDefault: () => { },
            target: {
                name: "addressLine2",
                value: "abcd",
            }
        };
        const wrapper = shallow(<CustomerProfile />);
        wrapper.setState({
            edit: true,
            readOnly: false,

        });
        const input = wrapper.find('.addressLine2');

        input.simulate('change', event);
        expect(input.length).toBe(1);
        expect(wrapper.state('addressLine2')).toBe("abcd");
    });

    it('on change of state field', () => {
        const event = {
            preventDefault: () => { },
            target: {
                name: "state",
                value: "def",
            }
        };
        const wrapper = shallow(<CustomerProfile />);
        wrapper.setState({
            edit: true,
            readOnly: false,

        });
        const input = wrapper.find('.state');

        input.simulate('change', event);
        expect(input.length).toBe(1);
        expect(wrapper.state('state')).toBe("def");
    });

    it('on change of country field', () => {
        const event = {
            preventDefault: () => { },
            target: {
                name: "country",
                value: "abcd",
            }
        };
        const wrapper = shallow(<CustomerProfile />);
        const input = wrapper.find('.country');
        input.simulate('change', event);
        expect(input.length).toBe(1);
        expect(wrapper.state('country')).toBe("abcd");
    });

    it('on change of pincode field', () => {
        const event = {
            preventDefault: () => { },
            target: {
                name: "pincode",
                value: "123456",
            }
        };
        const wrapper = shallow(<CustomerProfile />);
        const input = wrapper.find('.pincode');

        input.simulate('change', event);
        expect(input.length).toBe(1);
        expect(wrapper.state('pincode')).toBe("123456");
    });
    it('cancelProfile is called', () => {
        const wrapper = shallow(<CustomerProfile />);

        const event = {
            preventDefault: () => { },
        }

        wrapper.setState({
            edit: true
        });
        wrapper.update();
        const form = wrapper.find('.MyForm');
        wrapper.find('.RegisterButton').last().simulate('click', event);
        expect(wrapper.state('edit')).toEqual(false);
    });

    it('edit button on form 1', () => {
        const wrapper = shallow(<CustomerProfile />);
        const form = wrapper.find('.MyForm');
        const event = {
            preventDefault: () => { }
        }
        wrapper.setState({
            edit: false
        });
        wrapper.update();

        const button = form.find('.RegisterButton');
        button.simulate('click', event);

        expect(wrapper.state('edit')).toEqual(true);
    });


    it('edit function', () => {
        const wrapper = shallow(<CustomerProfile />);
        const form = wrapper.find('.MyForm');
        const event = {
            preventDefault: () => { }
        }
        wrapper.setState({
            edit: true
        });
        wrapper.update();
        const button = form.find('.RegisterButton').last();
        button.simulate('click', event);
        expect(wrapper.state('edit')).toEqual(false);
    });

    // it('on change of date of birth field', () => {
    //     const event = {
    //         preventDefault: () => { },
    //         target: {
    //             name: "dateOfBirth",
    //             value: "1997-12-12",
    //         }
    //     };
    //     const wrapper = shallow(<CustomerProfile />);
    //     const input = wrapper.find('#dob1');

    //     input.simulate('change', event);
    //     expect(input.length).toBe(1);
    //     expect(wrapper.state('dateOfBirth')).toBe("1997-12-12");
    // });

    // it('testing dateValidator() function' , () => {

    //     const event = {
    //         preventDefault: () => { },
    //         target: {
    //             name: "dateOfBirth",
    //             value: "1997-12-12",
    //         }
    //     };
    //     const wrapper = shallow(<CustomerProfile />);
    //     const input = wrapper.find('#dob1');

    //     input.simulate('change', event);
    //     expect(input.length).toBe(1);
    //     expect(wrapper.state('dateOfBirth')).toBe("1997-12-12");

        // const wrapper = shallow(<CustomerProfile />);
        // // wrapper.setState({dateOfBirth : '2017-12-12'});
        // const mockdateValidator = jest.fn(() => true);
        // var dateEntered = '2017-12-12';
        // expect(wrapper.instance().dateValidator()).toBe(true);
        // // expect(mockdateValidator).toHaveBeenCalledTimes(1);
		
   // })


})
