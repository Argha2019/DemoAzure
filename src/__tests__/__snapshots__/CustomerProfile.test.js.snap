// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer Profile render component 1`] = `
<CustomerTemplate>
  <div
    className="register-wrap"
  >
    <br />
    <br />
    <br />
    <br />
    <div
      className="register-html"
    >
      <h1
        className="fontWhite"
      >
        Profile
      </h1>
      <div
        className="Oms"
      >
        <br />
        <br />
        <Form
          className="MyForm register-form"
          inline={false}
        >
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "noGutters": false,
                },
                "render": [Function],
              }
            }
            controlId="username"
          >
            <Col
              className="alignLabel"
              md={3}
              sm={4}
            >
              <FormLabel
                className="label Labelling"
                column={false}
                srOnly={false}
              >
                Username
              </FormLabel>
            </Col>
            <Col
              md={9}
              sm={8}
            >
              <FormControl
                className="input"
                defaultValue=""
                name="username"
                readOnly={true}
                required={true}
                type="text"
              />
            </Col>
          </FormGroup>
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "noGutters": false,
                },
                "render": [Function],
              }
            }
            controlId="role"
          >
            <Col
              className="alignLabel"
              md={3}
              sm={4}
            >
              <FormLabel
                className="label Labelling"
                column={false}
                srOnly={false}
              >
                Role
              </FormLabel>
            </Col>
            <Col
              md={9}
              sm={8}
            >
              <FormControl
                className="input"
                name="role"
                readOnly={true}
                required={true}
                type="text"
              />
            </Col>
          </FormGroup>
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "noGutters": false,
                },
                "render": [Function],
              }
            }
            controlId="firstName"
          >
            <Col
              className="alignLabel"
              md={3}
              sm={4}
            >
              <FormLabel
                className="label Labelling"
                column={false}
                srOnly={false}
              >
                First Name
              </FormLabel>
            </Col>
            <Col
              md={9}
              sm={8}
            >
              <FormControl
                className="input firstName"
                defaultValue=""
                name="firstName"
                onChange={[Function]}
                readOnly={true}
                required={true}
                type="text"
              />
            </Col>
          </FormGroup>
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "noGutters": false,
                },
                "render": [Function],
              }
            }
            controlId="lastName"
          >
            <Col
              className="alignLabel"
              md={3}
              sm={4}
            >
              <FormLabel
                className="label Labelling"
                column={false}
                srOnly={false}
              >
                Last Name
              </FormLabel>
            </Col>
            <Col
              md={9}
              sm={8}
            >
              <FormControl
                className="input lastName"
                defaultValue=""
                name="lastName"
                onChange={[Function]}
                readOnly={true}
                required={true}
                type="text"
              />
            </Col>
          </FormGroup>
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "noGutters": false,
                },
                "render": [Function],
              }
            }
            controlId="dateOfBirth"
          >
            <Col
              className="alignLabel"
              md={3}
              sm={4}
            >
              <FormLabel
                className="label Labelling"
                column={false}
                srOnly={false}
              >
                Date of Birth
              </FormLabel>
            </Col>
            <Col
              md={9}
              sm={8}
            >
              <FormControl
                className="input dateOfBirth"
                id="dob1"
                name="dateOfBirth"
                onChange={[Function]}
                onFocus={[Function]}
                readOnly={true}
                required={true}
                type="text"
              />
            </Col>
          </FormGroup>
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "noGutters": false,
                },
                "render": [Function],
              }
            }
            controlId="phone"
          >
            <Col
              className="alignLabel"
              md={3}
              sm={4}
            >
              <FormLabel
                className="label Labelling"
                column={false}
                srOnly={false}
              >
                Phone
              </FormLabel>
            </Col>
            <Col
              md={9}
              sm={8}
            >
              <FormControl
                className="input phone"
                defaultValue=""
                name="phone"
                onChange={[Function]}
                readOnly={true}
                required={true}
                type="number"
              />
            </Col>
          </FormGroup>
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "noGutters": false,
                },
                "render": [Function],
              }
            }
            controlId="mail"
          >
            <Col
              className="alignLabel"
              md={3}
              sm={4}
            >
              <FormLabel
                className="label Labelling"
                column={false}
                srOnly={false}
              >
                Mail ID
              </FormLabel>
            </Col>
            <Col
              md={9}
              sm={8}
            >
              <FormControl
                className="input mail"
                defaultValue=""
                name="mail"
                onChange={[Function]}
                readOnly={true}
                required={true}
                type="email"
              />
            </Col>
          </FormGroup>
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "noGutters": false,
                },
                "render": [Function],
              }
            }
            controlId="addressLine1"
          >
            <Col
              className="alignLabel"
              md={3}
              sm={4}
            >
              <FormLabel
                className="label Labelling"
                column={false}
                srOnly={false}
              >
                Address Line 1
              </FormLabel>
            </Col>
            <Col
              md={9}
              sm={8}
            >
              <FormControl
                className="input addressLine1"
                defaultValue=""
                name="addressLine1"
                onChange={[Function]}
                readOnly={true}
                required={true}
                type="text"
              />
            </Col>
          </FormGroup>
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "noGutters": false,
                },
                "render": [Function],
              }
            }
            controlId="addressLine2"
          >
            <Col
              className="alignLabel"
              md={3}
              sm={4}
            >
              <FormLabel
                className="label Labelling"
                column={false}
                srOnly={false}
              >
                Address Line 2
              </FormLabel>
            </Col>
            <Col
              md={9}
              sm={8}
            >
              <FormControl
                className="input addressLine2"
                defaultValue=""
                name="addressLine2"
                onChange={[Function]}
                readOnly={true}
                required={true}
                type="text"
              />
            </Col>
          </FormGroup>
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "noGutters": false,
                },
                "render": [Function],
              }
            }
            controlId="state"
          >
            <Col
              className="alignLabel"
              md={3}
              sm={4}
            >
              <FormLabel
                className="label Labelling"
                column={false}
                srOnly={false}
              >
                State
              </FormLabel>
            </Col>
            <Col
              md={9}
              sm={8}
            >
              <FormControl
                className="input state"
                defaultValue=""
                name="state"
                onChange={[Function]}
                readOnly={true}
                required={true}
                type="text"
              />
            </Col>
          </FormGroup>
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "noGutters": false,
                },
                "render": [Function],
              }
            }
            controlId="country"
          >
            <Col
              className="alignLabel"
              md={3}
              sm={4}
            >
              <FormLabel
                className="label Labelling"
                column={false}
                srOnly={false}
              >
                Country
              </FormLabel>
            </Col>
            <Col
              md={9}
              sm={8}
            >
              <FormControl
                className="input country"
                defaultValue=""
                name="country"
                onChange={[Function]}
                readOnly={true}
                required={true}
                type="text"
              />
            </Col>
          </FormGroup>
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "noGutters": false,
                },
                "render": [Function],
              }
            }
            controlId="pincode"
          >
            <Col
              className="alignLabel"
              md={3}
              sm={4}
            >
              <FormLabel
                className="label Labelling"
                column={false}
                srOnly={false}
              >
                Pincode
              </FormLabel>
            </Col>
            <Col
              md={9}
              sm={8}
            >
              <FormControl
                className="input pincode"
                name="pincode"
                onChange={[Function]}
                readOnly={true}
                required={true}
                type="number"
              />
            </Col>
          </FormGroup>
          <div
            className="editButtonDiv"
          >
            <Button
              active={false}
              className="RegisterButton"
              disabled={false}
              onClick={[Function]}
              type="submit"
              variant="primary"
            >
              Edit
            </Button>
          </div>
        </Form>
      </div>
    </div>
  </div>
</CustomerTemplate>
`;
